### PR TITLE
COMMERCE-10175 Automation Test: Checkout products

### PR DESCRIPTION
**Pull request by Brunno0liveira**
https://issues.liferay.com/browse/COMMERCE-10175

### CODE REVIEW

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:stop

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:stop

**Comment by Brunno0liveira** [comment]
ci:stop

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:stop

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+				productUnitPrice = "15",
+				shippingMethod = "Standard");
+
+			AssertTextEquals(
+				locator1 = "CommerceAccelerators#CHECKOUT_SUCCESS_MESSAGE",
+				value1 = "Success! Your order has been processed.");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10306. Can checkout of an order of a product with price list with tier and promotion with bulk"
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase:181]: I think that "Can checkout of an order of a product..." can become "Can checkout an order of a product..."

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+				productUnitPrice = "30",
+				shippingMethod = "Standard");
+
+			AssertTextEquals(
+				locator1 = "CommerceAccelerators#CHECKOUT_SUCCESS_MESSAGE",
+				value1 = "Success! Your order has been processed.");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10307. Can checkout of an order of a product with price list with bulk and promotion with bulk"
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase:275]: I think that "Can checkout of an order of a product..." can become "Can checkout an order of a product..."

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+	@priority = "3"
+	test CanCheckoutProductWithTBulkPriceOnPromotionAndPriceList {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase:277]: For the test name **CanCheckoutProductWithTBulkPriceOnPromotionAndPriceList**, the "TBulk" what does it mean?

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+
+			AssertTextEquals(
+				locator1 = "CommerceAccelerators#CHECKOUT_SUCCESS_MESSAGE",
+				value1 = "Success! Your order has been processed.");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10281. Can checkout an order of a product with tier pricing only on the promotions"
+	@priority = "3"
+	test CanCheckoutProductWithTierPriceOnlyInPromotionTierPrice {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase:465]: Can you rename it in **CanCheckoutProductWithTierPriceOnlyInPromotion**?

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceDiscountCheckout.testcase [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+		CommerceConfiguration.commerceSetUp(minium="true");
+	}
+
+	tearDown {
+		CommerceConfiguration.commerceTearDown();
+	}
+
+	@description = "This is a test for COMMERCE-10343. Can checkout a product with two discounts, lower discount will be apply"
+	@priority = "3"
+	test CanCheckoutProductWithDiscountLowerDiscountIsApply {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceDiscountCheckout.testcase:19]: Can you rename it "CanCheckoutProductWithDiscountsLowerDiscountIsApplied"

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/paths/CommerceAccelerators.path [hunk 6]
```diff
@@ -160,11 +160,6 @@
 	<td>//*[@class='product-card-tiles']/div[${key_value}]</td>
 	<td></td>
 </tr>
-<tr>
-	<td>PRODUCT_CARD_DESCRIPTION</td>
-	<td>//p[@title='${key_productName}']/..//following-sibling::p[@class='card-body']//span[contains(normalize-space(text()),'${key_productDescription}')]</td>
-	<td></td>
-</tr>
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/paths/CommerceAccelerators.path:167]: Why this path it was removed and readded in another location in the same .path file?

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceDiscountCheckout.testcase [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+			
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "ABS Sensor");
+
+			CommerceAcceleratorsInitializer.viewDiscountedProductDetailsPage(
+				discountPercentage = "30",
+				productDiscountedPrice = "35.00",
+				productListPrice = "50.00");
+		}
+
+		task("And the higher discount is deleted so the lower discount can be apllied"){
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceDiscountCheckout.testcase]: Can you fix the "Apllied" to "Applied"?

Thanks!

**Review by pietrobocale — CHANGES_REQUESTED** [review]

**Comment by pietrobocale** [comment]
Review completed. Great work Brunno, take a look at the comments that I left, if you have some doubts please write me a Slack message

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase (hunk 3 above)
Comment (Brunno0liveira) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePromotionsCheckout.testcase:277]: The "T" was a typo the correct name of the test is **CanCheckoutProductWithBulkPriceOnPromotionAndPriceList**

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
Changes made

Thanks!

**Comment by Brunno0liveira** [comment]
ci:forward

**Pull request closed without merging**